apollo-router-core/src/query_planner/mod.rs

```diff
@@ -324,14 +324,17 @@ mod fetch {
 
             let query_span = tracing::info_span!("subfetch", service = service_name.as_str());
 
-            let Variables { variables, paths } = Variables::new(
-                &self.requires,
-                self.variable_usages.as_ref(),
-                data,
-                current_dir,
-                request,
-                schema,
-            )?;
+            let Variables { variables, paths } =
+                tracing::debug_span!("make_variables").in_scope(|| {
```

Comment:
:yarn: As per our ADR on tracing I think you would need to put that span as `#[instrument]` decorator on the `fn new()` of Variables.

I'm still not 100% sure if putting things all the time at the source is the best practice :thinking:

Reply:
I think the argument would be from which point of view we want to track. Is the caller naming spans for child function calls, or should a function always be traced the same way with the same span name?
It's not the issue here since `Variables::new` is used only there, and adding the attribute indeed looks better

Reply:
fixed in 2f93f39